module/metrics.go

```diff
@@ -36,6 +36,9 @@ type NetworkSecurityMetrics interface {
 
 	// OnRateLimitedUnicastMessage tracks the number of rate limited messages seen on the network.
 	OnRateLimitedUnicastMessage(role, msgType, topic, reason string)
+
+	// UnicastStreamDropped tracks the number of unicast streams that were dropped due to exceeding the limit.
+	UnicastStreamDropped()
```

Comment:
Should we rename this `UnicastStreamReset` ?

Reply:
we could, but it doesn't capture all resets, just streams that are reset due to exceeding the limit

Reply:
Seems we attempt on [closing](https://github.com/onflow/flow-go/blob/900b3188765f2a228284dd14d42749ba6dd25cd6/network/p2p/middleware/middleware.go#L472) (and not directly reset) upon meeting the limit. So `dropped` or `closed` may fit more than `reset`.

Reply:
it's actually [resetting](https://github.com/onflow/flow-go/blob/900b3188765f2a228284dd14d42749ba6dd25cd6/network/p2p/middleware/middleware.go#L477) since `success` is false. This is also confirmed by the unittests.